### ISSUE TITLE
Problem with xaml after upgrede to 0.9-rc1

### ISSUE THREAD

**Issue by OmidID** [AvaloniaUI/Avalonia#3340]
I get crash at Initialize method when I try to load Xaml file.
```
Avalonia.Markup.Xaml.XamlLoadException: 'No precompiled XAML found for ModulesSample.App, make sure to specify x:Class and include your XAML file as AvaloniaResource'
```
![image](https://user-images.githubusercontent.com/956077/70544548-ecba2580-1b74-11ea-9b6d-c03ef0a2be25.png)

I added `x:Class` to be sure it's right target.

```xml
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="ModulesSample.App">
  <Application.Styles>
    <StyleInclude Source="resm:Avalonia.Themes.Default.DefaultTheme.xaml?assembly=Avalonia.Themes.Default"/>
    <StyleInclude Source="resm:Avalonia.Themes.Default.Accents.BaseLight.xaml?assembly=Avalonia.Themes.Default"/>
  </Application.Styles>
</Application>
```

And this is what i have in .csproj file:

```xml
  <ItemGroup>
    <Compile Update="**\*.xaml.cs">
      <DependentUpon>%(Filename)</DependentUpon>
    </Compile>
    <AvaloniaResource Include="**\*.xaml">
      <SubType>Designer</SubType>
    </AvaloniaResource>
    <AvaloniaResource Include="Assets\*"/>
  </ItemGroup>
```

**Comment by donandren** [comment]
@OmidID i think you need to ensure you reference the same and only `Avalonia 0.9-*` nuget
and may be use` avares://` scheme for `<StyleInclude Source="avares://...`
like:
```xaml
        <StyleInclude Source="avares://Avalonia.Themes.Default/DefaultTheme.xaml"/>
        <StyleInclude Source="avares://Avalonia.Themes.Default/Accents/BaseLight.xaml"/>
```

**Comment by OmidID** [comment]
@donandren Actually I using build props files and I'm sure dependencies are OK.
You can get the source code from here and run the `ModulesSample` 
https://github.com/OmidID/Prism.Avalonia

**Comment by donandren** [comment]
try add **direct** nuget reference in the project where` app.xaml` is defined:
`<PackageReference Include="Avalonia" Version="0.9.0-rc.1" />` 
the avalonia build task responsible for generating resources might not be executing properly without direct reference to `Avalonia` nuget. 
Some time ago i've made a pr to fix and diagnose similar issues, but it's still not reviewed :( and still it's harder to diagnose without it.

**Comment by OmidID** [comment]
@donandren Great. that's the problem. Your changes are required.
After that I found out MainWindow needs the same approach with `x:Class`  and I found out another problem that I have no idea for it. Because in Prism all the Views will prepare by Resolver then dependencies will be injected in the constructor and constructor parameterless is required by Xaml compiler

https://github.com/OmidID/Prism.Avalonia/blob/dev-7.2.0/samples/ModulesSample/MainWindow.xaml.cs#L22

```
1>D:\Projects\Prism.Avalonia\samples\ModulesSample\MainWindow.xaml(1,2,1,2): Avalonia error XAMLIL: Unable to find public constructor for type ModulesSample:ModulesSample.MainWindow() (line 1 position 2) Line 1, position 2.
```

So I have no idea what is the solution. In case if we have 2 constructor resolver will use the default constructor.
